### PR TITLE
Fixes weapon circuit runtime and some slightly logging changes 

### PR DESCRIPTION
**Pull request by MNarath1**
## About The Pull Request

This PR fixes a runtime that occured for logging when it tried to call usr if the action was actually not called by a player like with a timer also fixes yet another runtime that occured when it tried to write into the non existing combat log of the assembly. This also includes a sligthly change to the gun circuit logging as the log message now includes the circuit creator too.
## Why It's Good For The Game

Runtimes bad. 
Runtimes that freeze the projectile in the air cause someone(me) forgot to consider timers as the action source of a weapon circuit firing are even worse.
## Changelog
:cl:
fix: weapon circuit runtime
tweak: adds creator to weapon circuit logging
/:cl:

### CODE REVIEW

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 1]
```diff
@@ -132,9 +132,10 @@
 	//Shooting Code:
 	A.preparePixelProjectile(target, src)
 	A.ignore_source_check = TRUE //needed else writing into firer will mess with the projectile collision
-	A.firer = "[usr.ckey]/[usr] trough [assembly]/[REF(assembly)] circuit"
+	A.firer = !usr ? "[assembly]/[REF(assembly)] circuit made by [assembly.creator]" : "[usr.ckey]/[usr] trough [assembly]/[REF(assembly)] circuit made by [assembly.creator]"
```
Comment (qwertyquerty) [inline code/modules/integrated_electronics/subtypes/manipulation.dm:135]: "trough"

**Review by qwertyquerty — APPROVED** [review]

File: code/modules/integrated_electronics/subtypes/manipulation.dm (hunk 1 above)
Comment (qwertyquerty) [reply code/modules/integrated_electronics/subtypes/manipulation.dm:135]: fix spelling

Comment (qwertyquerty) [inline code/modules/integrated_electronics/subtypes/manipulation.dm]: "trough"

**Review by qwertyquerty — DISMISSED** [review]
"trough"

File: code/modules/integrated_electronics/subtypes/manipulation.dm (hunk 1 above)
Comment (MNarath1) [reply code/modules/integrated_electronics/subtypes/manipulation.dm]: Ok is that better ?

**Comment by MNarath1** [comment]
@qwertyquerty is it better now?

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**